src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -376,6 +425,34 @@ bool CascadiaSettings::_IsPowerShellCoreInstalled(std::wstring_view programFileE
     return false;
 }
 
+// Function Description:
+// - Returns true if the executable for the distribution was found.
+// Arguments:
+// - A string that contains the distributions directory.
+// - A string representing the distinct WSL distribution.
+// - A ref of a path that receives the result of [distribution].exe's full path.
+// Return Value:
+// - true or false.
+bool CascadiaSettings::_IsWSLDistributionAvailable(std::filesystem::path WSLDirectory, std::wstring searchString, std::filesystem::path& cmdline)
+{
+    // step through all of the directory names to find the one with our search string
+    for (auto& directory : std::filesystem::directory_iterator(WSLDirectory)) {
+        std::wstring FileString = directory.path().wstring();
+        if (FileString.find(searchString) != std::wstring::npos) {
+
+            // step through all of the files looking for an executable.  
+            // I couldn't find a way of determining the validity of this step.
+            for (auto& file : std::filesystem::directory_iterator(directory.path())) {

```

Comment:
This seems plausible, but I would wait for the console team's feedback this coming week (hopefully) to see if there's a "supported" way of doing this.